### ISSUE TITLE
Cargo build fails

### ISSUE THREAD

**Issue by bpetit** [powercap/powercap-sys#1]
Hi,

Trying to use the library in a crate, with:

```
[dependencies]
powercap-sys = "0.1.0"
```

running cargo build I get:

```
   Compiling libc v0.2.79
   Compiling log v0.4.11
   Compiling cfg-if v0.1.10
   Compiling rustc-serialize v0.3.24
   Compiling term v0.4.6
   Compiling unicode-xid v0.0.4
   Compiling bitflags v0.8.2
   Compiling lazy_static v1.4.0
   Compiling memchr v2.3.3
   Compiling regex v0.2.11
   Compiling ucd-util v0.1.8
   Compiling bitflags v1.2.1
   Compiling libloading v0.4.3
   Compiling glob v0.2.11
   Compiling unicode-width v0.1.8
   Compiling utf8-ranges v1.0.4
   Compiling strsim v0.8.0
   Compiling vec_map v0.8.2
   Compiling ansi_term v0.11.0
   Compiling bitflags v0.9.1
   Compiling lazy_static v0.2.11
   Compiling peeking_take_while v0.1.2
   Compiling pkg-config v0.3.19
   Compiling thread_local v0.3.6
   Compiling regex-syntax v0.5.6
   Compiling textwrap v0.11.0
   Compiling clang-sys v0.19.0
   Compiling syntex_pos v0.58.1
   Compiling log v0.3.9
   Compiling aho-corasick v0.6.10
   Compiling syntex_errors v0.58.1
   Compiling memchr v1.0.2
   Compiling atty v0.2.14
   Compiling nom v3.2.1
   Compiling clap v2.33.3
   Compiling syntex_syntax v0.58.1
error[E0423]: expected function, tuple struct or tuple variant, found struct `ast::Name`
   --> /home/bpetit/.cargo/registry/src/github.com-1ecc6299db9ec823/syntex_syntax-0.58.1/src/symbol.rs:146:27
    |
146 |                       name: ast::Name($index),
    |                             ^^^^^^^^^
...
165 | / declare_keywords! {
166 | |     // Invalid identifier
167 | |     (0,  Invalid,        "")
168 | |
...   |
231 | |     (56, CrateRoot, "{{root}}")
232 | | }
    | |_- in this macro invocation

error: aborting due to previous error

For more information about this error, try `rustc --explain E0423`.
error: could not compile `syntex_syntax`.
warning: build failed, waiting for other jobs to finish...
error: build failed
```

I'm on Ubuntu 18.04 and I've installed libpowercap0 libpowercap-dev powercap-utils prior to that.

Any idea about what I'm missing ?

Thanks a lot

**Comment by connorimes** [comment]
Thanks for the report.  This appears to be a failure in a another crate - `syntex_syntax` which is a transitive build dependency of `bindgen`, but is no longer maintained.  I see the same error with rust 1.47.0.

I will update the build dependency versions.

**Comment by connorimes** [comment]
If you're able to test this out before I tag and push a release to crates.io, that would be helpful.  I don't really work in Rust anymore and haven't been using these -sys bindings.

**Comment by bpetit** [comment]
Running cargo build with:

```
powercap-sys = { git = "https://github.com/powercap/powercap-sys.git" }
```

I don't have the same error anymore. Now I have:

```
--- stderr
/usr/include/unistd.h:229:10: fatal error: 'stddef.h' file not found
/usr/include/unistd.h:229:10: fatal error: 'stddef.h' file not found, err: true
thread 'main' panicked at 'Unable to generate bindings: ()', /home/bpetit/.cargo/git/checkouts/powercap-sys-7f8c3f5ea7733edf/86a6eb3/build.rs:19:20
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```
But the previous steps are okay.
So I'd say the error I had is fixed with your changes.

Thanks !

**Comment by connorimes** [comment]
Great, thank you for testing.  I've tagged and published v0.2.0 to crates.io.

I think what you're seeing now is a clang-related error, about which I can only speculate without knowing more about your build environment.  If you think this is still a powercap-sys problem, please feel free to open another issue and we'll address it there.  Cheers.
